.github/workflows/linux-eic-shell.yml

```diff
@@ -135,12 +135,15 @@ jobs:
   clang-tidy-iwyu:
     runs-on: ubuntu-latest
     needs: build
+    container:
+      # provides IWYU 8.21
+      image: ubuntu:24.04
     steps:
+    - run: apt update && apt install -y clang-tidy git iwyu zstd
     - run: echo "FETCH_DEPTH=$(( ${{ github.event.pull_request.commits }} + 1 ))" >> "${GITHUB_ENV}"
     - uses: actions/checkout@v4
       with:
         fetch-depth: ${{ env.FETCH_DEPTH }}
```

Comment:
OT but at some point we should just switch to  `actions/checkout` with `filter: "tree:0"` instead of determining FETCH_DEPTH.